### PR TITLE
converted basenominaltransformer to polars

### PR DESCRIPTION
**Pull request by limlam96**
edited transformer and tests to use narwhals framework

https://github.com/lvgig/tubular/issues/334

### CODE REVIEW

**Review by davidhopkinson26 — CHANGES_REQUESTED** [review]
think we also need to edit inverse_transform.  It has a replace operation in there.

File: tests/nominal/test_BaseNominalTransformer.py [hunk 1]
```diff
@@ -20,22 +21,38 @@ class GenericNominalTransformTests(GenericTransformTests):
...
+        self,
+        initialized_transformers,
+        library,
+    ):
         """Test an exception is raised if non-mappable rows are present in X."""
-        df = d.create_df_1()
+        df = d.create_df_1(library=library)
 
-        x = initialized_transformers[self.transformer_name]
+        transformer = initialized_transformers[self.transformer_name]
```
Comment (davidhopkinson26) [inline tests/nominal/test_BaseNominalTransformer.py:47]: P: thanks for changing from x -> transformer!

**Comment by limlam96** [comment]
> think we also need to edit inverse_transform. It has a replace operation in there.

inverse_transform is in 'NominalToIntegerTransformer' so was leaving for that issue? unless i'm mixed up!

**Comment by davidhopkinson26** [comment]
> > think we also need to edit inverse_transform. It has a replace operation in there.
> 
> inverse_transform is in 'NominalToIntegerTransformer' so was leaving for that issue? unless i'm mixed up!

My bad sorry!

**Review by davidhopkinson26 — APPROVED** [review]

**Pull request merged**